sdk/resource/auto_test.go

```diff
@@ -28,32 +29,49 @@ import (
 
 func TestDetect(t *testing.T) {
 	cases := []struct {
-		name             string
-		schema1, schema2 string
-		isErr            bool
+		name    string
+		schema  []string
+		wantErr error
 	}{
 		{
-			name:    "different schema urls",
-			schema1: "https://opentelemetry.io/schemas/1.3.0",
-			schema2: "https://opentelemetry.io/schemas/1.4.0",
-			isErr:   true,
+			name: "two different schema urls",
+			schema: []string{
+				"https://opentelemetry.io/schemas/1.3.0",
+				"https://opentelemetry.io/schemas/1.4.0",
+			},
+			wantErr: resource.ErrSchemaURLConflict,
 		},
 		{
-			name:    "same schema url",
-			schema1: "https://opentelemetry.io/schemas/1.4.0",
-			schema2: "https://opentelemetry.io/schemas/1.4.0",
-			isErr:   false,
+			name: "three different schema urls",
+			schema: []string{
+				"https://opentelemetry.io/schemas/1.3.0",
+				"https://opentelemetry.io/schemas/1.4.0",
+				"https://opentelemetry.io/schemas/1.5.0",
+			},
+			wantErr: resource.ErrSchemaURLConflict,
+		},
+		{
+			name: "same schema url",
```

Comment:
We are not checking if the returned resource has schema URL equal to `https://opentelemetry.io/schemas/1.4.0`.

Maybe simply adding `wantSchemaURL`?

We could then avoid having following logic in test 

```go
if errors.Is(c.wantErr, resource.ErrSchemaURLConflict) {
	assert.Equal(t, "", r.SchemaURL())
```

and simply have

```go
assert.Equal(t, c.wantSchemaURL, r.SchemaURL())
```

Reply:
Sure

Reply:
Updated.